http/jaxrs-reactive/src/test/java/io/quarkus/ts/http/jaxrs/reactive/MediaTypeSelectionIT.java

```diff
@@ -0,0 +1,165 @@
+package io.quarkus.ts.http.jaxrs.reactive;
+
+import static io.quarkus.ts.http.jaxrs.reactive.MediaTypeResource.GET_WITHOUT_CONSUMED_MEDIA_TYPES;
+import static io.quarkus.ts.http.jaxrs.reactive.MediaTypeResource.MEDIA_TYPE_PATH;
+import static io.quarkus.ts.http.jaxrs.reactive.MediaTypeResource.PATCH_APP_OCTET_STREAM;
+import static io.quarkus.ts.http.jaxrs.reactive.MediaTypeResource.PATCH_WITHOUT_CONSUMED_MEDIA_TYPES;
+import static io.quarkus.ts.http.jaxrs.reactive.MediaTypeResource.POST_APP_ATOM_XML_AND_TEXT_XML_EXT_AND_APP_JSON_PATCH_JSON;
+import static io.quarkus.ts.http.jaxrs.reactive.MediaTypeResource.POST_APP_JSON_AND_TEXT_XML_AND_TEXT_PLAIN;
+import static javax.ws.rs.core.MediaType.APPLICATION_ATOM_XML;
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_PATCH_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_SVG_XML;
+import static javax.ws.rs.core.MediaType.APPLICATION_XHTML_XML;
+import static javax.ws.rs.core.MediaType.APPLICATION_XML;
+import static javax.ws.rs.core.MediaType.TEXT_XML;
+
+import javax.ws.rs.core.MediaType;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+@QuarkusScenario
+public class MediaTypeSelectionIT {
+
+    @Test
+    public void testUnsupportedMediaType() {
+        // no resource method consumes multipart/form-data nor wild card
+        RestAssured.given()
+                .contentType("multipart/form-data")
+                .post(MEDIA_TYPE_PATH)
+                .then()
+                .statusCode(415);
+    }
+
+    @Test
+    public void testExactMatchApplicationSoapXml() {
+        RestAssured.given()
+                .contentType("application/soap+xml")
+                .post(MEDIA_TYPE_PATH)
+                .then()
+                .statusCode(200)
+                .body(Matchers.equalTo("application/soap+xml"));
+    }
+
+    @Test
+    public void testMediaSubTypesRanges() {
```

Comment:
`testAcceptHeaders` has comments above each RestAssured invocation, similar context should be provided here

Maybe each RestAssured invocation could be separate test method (applies to both `testMediaSubTypesRanges` and `testAcceptHeaders`). Or maybe 2 test methods, one for POST_APP_ATOM_XML_AND_TEXT_XML_EXT_AND_APP_JSON_PATCH_JSON and one for POST_APP_JSON_AND_TEXT_XML_AND_TEXT_PLAIN items.
Keeping this as optional rewuest, feel free to decide on your own.

Reply:
added comments, thx